rskj-core/src/main/java/co/rsk/mine/BlockToMineBuilder.java

```diff
@@ -89,7 +90,7 @@ public BlockToMineBuilder(
         this.difficultyCalculator = Objects.requireNonNull(difficultyCalculator);
         this.gasLimitCalculator = Objects.requireNonNull(gasLimitCalculator);
         this.validationRules = Objects.requireNonNull(validationRules);
-
+        this.isRegtest = config.getBlockchainConfig() instanceof RegTestConfig;
```

Comment:
I think it would be better to build an instance of `BlockToMineBuilder` in a bean factory method and, according to the `config.netName()`, inject a _strategy_ for this regtest behavior (`CurrentTimeResolver` interface maybe?)